Formula/esptool.rb

```diff
@@ -0,0 +1,35 @@
+class Esptool < Formula
+  include Language::Python::Virtualenv
+
+  desc "ESP8266 and ESP32 serial bootloader utility"
+  homepage "https://github.com/espressif/esptool"
+  url "https://github.com/espressif/esptool/archive/v2.5.1.tar.gz"
+  sha256 "6730b69d8ba09ff09f845873eaeef352bd18a4ee24b31b0898fd8fc5f178c037"
+
+  depends_on "python"
+
+  resource "ecdsa" do
+    url "https://files.pythonhosted.org/packages/f9/e5/99ebb176e47f150ac115ffeda5fedb6a3dbb3c00c74a59fd84ddf12f5857/ecdsa-0.13.tar.gz"
+    sha256 "64cf1ee26d1cde3c73c6d7d107f835fed7c6a2904aef9eac223d57ad800c43fa"
+  end
+
+  resource "pyaes" do
+    url "https://files.pythonhosted.org/packages/44/66/2c17bae31c906613795711fc78045c285048168919ace2220daa372c7d72/pyaes-1.6.1.tar.gz"
+    sha256 "02c1b1405c38d3c370b085fb952dd8bea3fadcee6411ad99f312cc129c536d8f"
+  end
+
+  resource "pyserial" do
+    url "https://files.pythonhosted.org/packages/cc/74/11b04703ec416717b247d789103277269d567db575d2fd88f25d9767fe3d/pyserial-3.4.tar.gz"
+    sha256 "6e2d401fdee0eab996cf734e67773a0143b932772ca8b42451440cfed942c627"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match(/#{version}/, shell_output("#{bin}/esptool.py version"))
```

Comment:
Can we create a test that checks the functionality of the software instead of this ?

Reply:
Unfortunately this software requires specialized hardware to do proper checks (e.g. image flashing, flash erasing) . I've tried coming up with something better than this, but I'm afraid it's the best we've got for now.

Reply:
Try performing an operation and check that it fails.

Reply:
Wasn't there some function to transform images that could be used ?

Reply:
@javian Indeed, but those binaries are quite big. Including them in the formula as a base64-encoded string seems pointless. However, there seem to be very small hello world binaries here: https://github.com/espressif/esptool/tree/master/test/images so I took one of them to test image reading functionality of this tool.